lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2661,12 +2661,19 @@ void IRGenDebugInfoImpl::addFailureMessageToCurrentLoc(IRBuilder &Builder,
   else {
     std::string FuncName = "Swift runtime failure: ";
     FuncName += failureMsg;
-    llvm::DIFile *File = getOrCreateFile({}, {});
+    // CodeView consumers do not correctly handle an artificially generated
+    // file, thus use the original location's file as the file for the debug
+    // function, and prevent reuse of this debug function.
+    bool useCompilerGeneratedFile =
+        Opts.DebugInfoFormat != IRGenDebugInfoFormat::CodeView;
```

Comment:
can you add a `bool generateForCodeView()` (better name welcome) and use that for all the comparisons? This will make it clearer that these are all workarounds for CodeView limitations, and make it easy to grep for them.